grunt/robots.js

```diff
@@ -1,32 +1,25 @@
+"use strict";
+
 module.exports = function ( grunt )
 {
-    "use strict";
-
-    grunt.registerTask( "envRobotsIndex", "Set robots.txt to index the site", function ()
-    {
-        grunt.config( "useRobotsIndex", true );
-    } );
-
-    grunt.registerTask( "envRobotsNoIndex", "Set robots.txt to NOT index the site", function ()
-    {
-        grunt.config( "useRobotsIndex", false );
-    } );
+    var env = grunt.config( "envName" );
+    var host = grunt.config( "env" ).host;
 
     grunt.registerTask( "robots", "Write a robots.txt", function ()
     {
-        if( grunt.config( "useRobotsIndex" ) && grunt.config( "Host" ) )
-        {
-            grunt.log.write( "Writing robots.txt for indexing" );
-            grunt.file.write( "build/robots.txt",
-                "User-agent: *\n"
-                + "Sitemap: " + grunt.config( "Host" ) + "/build/sitemap.txt\n" );
-        }
-        else
+        if( env !== "prod" || !host  )
         {
             grunt.log.write( "Writing robots.txt for NO indexing" );
             grunt.file.write( "build/robots.txt",
                 "User-agent: *\n"
                 + "disallow: /\n" );
+            return;
         }
+
+        // TODO: can we determine if there is a sitemap file or not?
```

Comment:
There should probably always be a robots.txt, but if there is no sitemap then it shouldn't write that line out. I don't know enough about sitemaps to say if it is better to just have a default sitemap.txt that points at the index file so we never have the no-content issue but that map is incomplete, or we shouldn't include sitemap.txt unless the content of it is relatively complete. To be fair, we should actually be maintaining this file for all the projects where it is relevant anyway.